mlflow/telemetry/client.py

```diff
@@ -81,18 +81,21 @@ def _send_batch(self):
             # TODO: record this case
             _logger.debug("Telemetry queue is full, skipping sending data.")
 
-    def _process_records(self, records: list[TelemetryRecord]):
+    def _process_records(self, records: list[APIRecord]):
         """Process a batch of telemetry records."""
         try:
-            data = {
-                "records": [
-                    {"data": record.data, "partition-key": record.partition_key}
-                    for record in records
-                ]
-            }
+            telemetry_info = self._get_telemetry_info_dict()
```

Comment:
I moved this inside _process_records so that we can suppress logs generated during this call. In addition, I feel like we don't need to check this too often.